### PR TITLE
Divisible and Decidable

### PR DESCRIPTION
**Pull request by paf31**
WIP. Still need to add comments and update the LICENSE file.

### CODE REVIEW

**Comment by garyb** [comment]
Looks good to me, I'm not really that familiar with these classes though, so don't have any deep insights :wink:. Comments would definitely be good.

I assume this doesn't break anything in the dependency hierarchy either, I don't think much depends on `contravariant` currently.

**Comment by garyb** [comment]
`const` and `lens` actually, that's it.

**Comment by paf31** [comment]
Ready for review.

**Comment by garyb** [comment]
:+1: 

I have a feeling I defined `Op` somewhere else, but I can't find it now. Perhaps it was just in something I was experimenting with.

File: src/Data/Functor/Contravariant/Divisible.purs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+divided :: forall f a b. (Divide f) => f a -> f b -> f (Tuple a b)
+divided = divide id
+
+-- | `Decide` is the contravariant analogue of `Alt`.
+class (Divide f) <= Decide f where
+  decide :: forall a b c. (a -> Either b c) -> f b -> f c -> f a
+  
+-- | `Decidable` is the contravariant analogue of `Alternative`.
+class (Decide f, Divisible f) <= Decidable f where
+  lose :: forall a b c. (forall r. a -> r) -> f a
```
Comment (xuwei-k) [inline src/Data/Functor/Contravariant/Divisible.purs:90]: type parameter `b` and `c` unused?

**Pull request merged**